### PR TITLE
Update release.md to reflect notes taken during 3.5.15 release

### PR DESCRIPTION
**Pull request by wenjiaswe**
Update release.md based on 3.5.15 release: https://github.com/etcd-io/etcd/issues/18247#issuecomment-2240080965

Please read https://github.com/etcd-io/etcd/blob/main/CONTRIBUTING.md#contribution-flow.

### CODE REVIEW

**Review by serathius — APPROVED** [review]

File: Documentation/contributor-guide/release.md [hunk 1]
```diff
@@ -62,6 +62,7 @@ which don't need to be executed before releasing each version.
 4. Authenticate the image registry, refer to [Authentication methods](https://cloud.google.com/container-registry/docs/advanced-authentication).
    - `gcloud auth login`
    - `gcloud auth configure-docker`
+5. Ensure there is at least 500MB of free space on your Linux machine.
```
Comment (ahrtr) [inline Documentation/contributor-guide/release.md:65]: Could you please follow https://github.com/etcd-io/etcd/pull/18346#discussion_r1685265865?

Comment (wenjiaswe) [reply Documentation/contributor-guide/release.md]: Oops, did that one wrong... Thanks! Updated.

**Comment by ahrtr** [comment]
Please try not to create branch on the etcd-io/etcd directly (e.g. [wenjiaswe-patch-1](https://github.com/etcd-io/etcd/tree/wenjiaswe-patch-1) in this PR). Instead, please fork the repo, and create branch on your forked repo.

**Review by ahrtr — APPROVED** [review]

**Comment by ivanvc** [comment]
> Please try not to create branch on the etcd-io/etcd directly (e.g. [wenjiaswe-patch-1](https://github.com/etcd-io/etcd/tree/wenjiaswe-patch-1) in this PR). Instead, please fork the repo, and create branch on your forked repo.

Maybe it is a good time to clean up branches `wenjiaswe-patch-1` and `ptabor-patch-1`?

**Comment by jmhbnz** [comment]
> Maybe it is a good time to clean up branches `wenjiaswe-patch-1` and `ptabor-patch-1`?

Done - good to clean up.

**Pull request merged**